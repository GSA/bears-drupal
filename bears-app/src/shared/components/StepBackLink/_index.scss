#usagov-bears-app {
  .step-back-link {
    color: #979797;
  }
}
