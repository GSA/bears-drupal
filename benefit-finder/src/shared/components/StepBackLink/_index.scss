#benefit-finder {
  .step-back-link {
    color: #979797;
  }
}
