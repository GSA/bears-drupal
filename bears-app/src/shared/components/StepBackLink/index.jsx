import PropTypes from 'prop-types'
import { Button } from '../index'
import './_index.scss'

/**
 * a functional component that supports backwards navigation in the step indicator component
 * @component
 * @param {function} setCurrent - inherited state function
 * @param {number} currentIndex - inherited current index state
 * @return {html} returns markup for a usa unstyled button
 */
const StepBackLink = ({ setCurrent, currentIndex }) => {
  /**
   * a handler that manages which index is updated on the click event
   * @function
   * @return {number} returns new index state for current
   */
  const handleBack = (setCurrent, currentIndex) => {
    currentIndex === 0 ? setCurrent(0) : setCurrent(currentIndex - 1)
  }

  return (
    <Button
      className="step-back-link"
      unstyled
      onClick={() => handleBack(setCurrent, currentIndex)}
    >
      Back
    </Button>
  )
}

StepBackLink.propTypes = {
  setCurrent: PropTypes.func,
  currentIndex: PropTypes.number,
}

export default StepBackLink
