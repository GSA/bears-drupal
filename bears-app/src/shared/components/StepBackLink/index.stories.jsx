import StepBackLink from './index.jsx'

export default {
  component: StepBackLink,
  tags: ['autodocs'],
  args: {
    setCurrent: currentIndex =>
      currentIndex === 0
        ? alert('back to index 0')
        : alert(`back to index ${currentIndex}`),
    currentIndex: 0,
  },
}

export const Primary = {}
