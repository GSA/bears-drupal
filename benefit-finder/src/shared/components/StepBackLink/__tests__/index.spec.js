import renderer from 'react-test-renderer'
import StepBackLink from '../index.jsx'

describe('StepBackLink', () => {
  it('renders a match to the previous snapshot', () => {
    const component = renderer.create(<StepBackLink />)
    expect(component.toJSON()).toMatchSnapshot()
  })
})
