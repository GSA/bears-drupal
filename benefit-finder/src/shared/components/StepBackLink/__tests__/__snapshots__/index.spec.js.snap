// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StepBackLink renders a match to the previous snapshot 1`] = `
<button
  className=""
  onClick={[Function]}
  type="button"
>
  Back
</button>
`;
